reactive/docker.py

```diff
@@ -131,20 +131,23 @@ def install():
     runtime = determine_apt_source()
     remove_state('nvidia-docker.supported')
     remove_state('nvidia-docker.installed')
-    if runtime == 'upstream':
-        install_from_upstream_apt()
-    elif runtime == 'nvidia':
+    if runtime == 'nvidia':
         set_state('nvidia-docker.supported')
         install_from_nvidia_apt()
         set_state('nvidia-docker.installed')
-    elif runtime == 'apt':
-        install_from_archive_apt()
-    elif runtime == 'custom':
-        if not install_from_custom_apt():
-            return False  # If install fails, stop.
+        docker.set_daemon_json('default-runtime', 'nvidia')
     else:
-        hookenv.log('Unknown runtime {}'.format(runtime))
-        return False
+        docker.delete_daemon_json('default-runtime')
+        if runtime == 'upstream':
+            install_from_upstream_apt()
+        elif runtime == 'apt':
+            install_from_archive_apt()
+        elif runtime == 'custom':
+            if not install_from_custom_apt():
+                return False  # If install fails, stop.
+        else:
+            hookenv.log('Unknown runtime {}'.format(runtime))
+            return False
```

Comment:
This looks a little confusing, but it's just a reordering of the if statement so I could say, "if it's nvidia, set the default-runtime; if it's anything else, clear the default-runtime."